app/models/foreman_snapshot_management/proxmox_extensions.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+module ForemanSnapshotManagement
+  module ProxmoxExtensions
+    # Extend Proxmox's capabilities with snapshots.
+    def capabilities
+      super + [:snapshots]
+    end
+
+    # Create a Snapshot.
+    #
+    # This method creates a Snapshot with a given name and optional description.
+    def create_snapshot(host, name, description, _include_ram = false)
+      server = find_vm_by_uuid host.uuid
+      raise 'Name must contain atleast 2 characters starting with alphabet. Valid characters are A-Z a-z 0-9 _' unless name =~ /^[A-Za-z][\w]{1,}$/
```

Comment:
I guess it should be "at least" instead of "atleast"

Reply:
The validation should happen for the update case, too.

Reply:
Also this should be translateable, because it is presented on the GUI (see https://projects.theforeman.org/projects/foreman/wiki/Translating#Exceptions)

Reply:
> The validation should happen for the update case, too.

Proxmox snapshot name cannot be updated since snapname == snapshot_id. So, it doesn't need to raise validation exception. Also, if user tries to update snapshot name it raises exception that name is not modifiable.